staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -158,11 +158,16 @@ func diffMaps(original, modified map[string]interface{}, t reflect.Type, ignoreC
 		switch originalValueTyped := originalValue.(type) {
 		case map[string]interface{}:
 			modifiedValueTyped := modifiedValue.(map[string]interface{})
-			fieldType, _, _, err := forkedjson.LookupPatchMetadata(t, key)
+			fieldType, fieldPatchStrategy, _, err := forkedjson.LookupPatchMetadata(t, key)
 			if err != nil {
 				return nil, err
 			}
 
+			if fieldPatchStrategy == replaceDirective {
```

Comment:
trying to reason through what ignoreChangesAndAdditions and ignoreDeletions would mean to this